TicketSystem/TicketSystem.DAL/Repositories/GenericRepository.cs

```diff
@@ -28,8 +29,10 @@ public async Task UpdateAsync(TEntity item, CancellationToken cancellationToken)
         await _context.SaveChangesAsync(cancellationToken);
     }
 
-    public async Task RemoveAsync(TEntity item, CancellationToken cancellationToken)
+    public async Task RemoveAsync(int id, CancellationToken cancellationToken)
     {
+        var item = await _dbSet.FirstAsync(x => x.Id == id, cancellationToken: cancellationToken);
```

Comment:
FirstOrDefault will be better 
If we already don't have this resource - good news, we already done this job :)